src/build.d

```diff
@@ -533,6 +534,55 @@ alias zip = memoize!(function()
     return new DependencyRef(dep);
 });
 
+string d2html(string sourceFile)
+{
+    auto ext = sourceFile.extension();
+    assert(ext == ".d" || ext == ".di", sourceFile);
+    const htmlFilePrefix = (sourceFile.baseName == "package.d") ?
+        sourceFile[0 .. $ - "package.d".length - 1] :
+        sourceFile[0 .. $ - ext.length];
+    return htmlFilePrefix ~ ".html";
+}
+
+alias html = memoize!(function()
+{
+    Dependency htmlDep;
+    htmlDep.name = "html";
+    htmlDep.description = "Generate html docs, requires DMD and STDDOC to be set";
+    auto docSources = .sources.root ~ .sources.lexer ~ .sources.dmd ~ env["D"].buildPath("frontend.d");
+    if (env.get("DMD", null).length == 0) {
+        htmlDep.commandFunction = () {
+            writefln("ERROR: cannot build '%s' unless DMD is set", htmlDep.name);
+        };
+    } else {
+        htmlDep.msg = "HTML " ~ docSources.join(" ");
+        htmlDep.deps = new DependencyRef[docSources.length];
+        const stddocs = env.get("STDDOC", "").split();
+        foreach(i; 0 .. docSources.length) {
+            Dependency docDep;
```

Comment:
Maybe split this off into another local function and declare the dependencies via an UFCS chain?

Reply:
What's nice about the current implementation is that I can create slices into the docSources array so I don't have to allocate a new array for each dependency.  Using a UFCS chain would make that difficult.

Reply:
Slicing instead of reallocating is indeed a good idea. You could keep it by using `docSources.slide(1)`